File: 백준/Bronze/2920. 음계/음계.c
#include <stdio.h>

int main() {
    int a[8];
    int i;
    

    for(i = 0; i < 8; i++) {
        scanf("%d", &a[i]);
    }


    if(a[0]==1 && a[1]==2 && a[2]==3 && a[3]==4 &&
       a[4]==5 && a[5]==6 && a[6]==7 && a[7]==8) {
        printf("ascending\n");
    }

    else if(a[0]==8 && a[1]==7 && a[2]==6 && a[3]==5 &&
            a[4]==4 && a[5]==3 && a[6]==2 && a[7]==1) {
        printf("descending\n");
    }

    else {
        printf("mixed\n");
    }

    return 0;
}

File: 백준/Bronze/2920. 음계/음계.py
a = input().split()
b = len(a)
c = 0
for i in range(b - 7):
    if(a[i] == '1' and a[i+1] == '2' and a[i+2] == '3' and a[i+3] == '4' and 
       a[i+4] == '5' and a[i+5] == '6' and a[i+6] == '7' and a[i+7] == '8'):
        c = 1
    elif(a[i] == '8' and a[i+1] == '7' and a[i+2] == '6' and a[i+3] == '5' and 
         a[i+4] == '4' and a[i+5] == '3' and a[i+6] == '2' and a[i+7] == '1'):
        c = -1

if(c == 1):
    print('ascending')
elif(c == -1):
    print('descending')
else:
    print('mixed')
